### PR TITLE
Add NodeSelector Config for k8s array

### PR DESCRIPTION
**Pull request by migueltol22**
# TL;DR
Add support for specifying node selector as part of k8s array plugin config

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/220

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: go/tasks/plugins/array/k8s/launcher.go [hunk 1]
```diff
@@ -52,6 +52,14 @@ func ApplyPodPolicies(_ context.Context, cfg *Config, pod *corev1.Pod) *corev1.P
 	return pod
 }
 
+func applyNodeSelectorLabels(_ context.Context, cfg *Config, pod *corev1.Pod) *corev1.Pod {
```
Comment (vsbus) [inline go/tasks/plugins/array/k8s/launcher.go]: should function start with uppercase? : `ApplyNodeSelectorLabels`

**Comment by surindersinghp** [comment]
please add some tests

File: go/tasks/plugins/array/k8s/config.go [hunk 2]
```diff
@@ -32,13 +32,19 @@ var (
 	configSection = config.MustRegisterSection(configSectionKey, defaultConfig)
 )
 
+type NodeSelectorConfig struct {
+	key   string `json:"key", pflag:",Node Selector key.`
```
Comment (EngHabu) [inline go/tasks/plugins/array/k8s/config.go]: I think Pod's NodeSelector supports multiple keys here right? and it does an AND between them or something?
we should match that behavior then

Comment (surindersinghp) [reply go/tasks/plugins/array/k8s/config.go]: agreed, like we do for interruptible: https://github.com/lyft/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/config/config.go#L51

Comment (migueltol22) [reply go/tasks/plugins/array/k8s/config.go]: Thanks for the reference! Updated.

File: go/tasks/plugins/array/k8s/launcher.go (hunk 1 above)
Comment (migueltol22) [reply go/tasks/plugins/array/k8s/launcher.go]: Since I'm not exporting this function, it should be fine lowercase.

Comment (vsbus) [reply go/tasks/plugins/array/k8s/launcher.go]: ah, gotcha! 👍

**Pull request merged**